### PR TITLE
Use head_branch metric

### PR DESCRIPTION
**Pull request by th-le**
Follow up for https://github.com/actions/actions-runner-controller/issues/2176.

This PR uses the branch_name for metrics

### CODE REVIEW

**Comment by th-le** [comment]
@mumoshu could you help take a look

File: go.mod [hunk 1]
```diff
@@ -60,6 +62,7 @@ require (
 	github.com/golang/protobuf v1.5.2 // indirect
 	github.com/google/gnostic v0.5.7-v3refs // indirect
 	github.com/google/go-github/v45 v45.2.0 // indirect
+	github.com/google/go-github/v52 v52.0.0
```
Comment (mumoshu) [inline go.mod]: Thank you @th-le! TIL that this had to be bumped in order to accommodate https://github.com/google/go-github/commit/9385ab02991f311db7d6be6a67717d0ff50fc84f.
However, I'm still unsure why this had to be in this second block of `require`.
Shouldn't it have resulted in updating L13 of this file?!

File: go.mod [hunk 2]
```diff
@@ -26,23 +25,25 @@ require (
 	github.com/teambition/rrule-go v1.8.2
 	go.uber.org/multierr v1.7.0
 	go.uber.org/zap v1.24.0
-	golang.org/x/net v0.8.0
-	golang.org/x/oauth2 v0.1.0
+	golang.org/x/net v0.9.0
+	golang.org/x/oauth2 v0.7.0
```
Comment (mumoshu) [inline go.mod:29]: These two versions seem to align with go-github v52 👍 

https://github.com/google/go-github/blob/v52.0.0/go.mod

File: go.mod [hunk 3]
```diff
@@ -26,23 +25,25 @@ require (
...
 	golang.org/x/sync v0.1.0
 	gomodules.xyz/jsonpatch/v2 v2.2.0
 	gopkg.in/yaml.v2 v2.4.0
 	k8s.io/api v0.26.2
 	k8s.io/apimachinery v0.26.2
 	k8s.io/client-go v0.26.2
-	sigs.k8s.io/controller-runtime v0.14.4
+	sigs.k8s.io/controller-runtime v0.14.6
```
Comment (mumoshu) [inline go.mod]: Can we leave this controller-runtime upgrade to another pull request to make this pull request focused on the head_branch addition? @th-le

**Review by mumoshu — APPROVED** [review]
I made changes that I thought necessary myself and merging this now!
It would be great if you could confirm when you have time ☺️ 
Thanks again for your contribution @th-le!!

**Comment by th-le** [comment]
the changes look good. thank you so much @mumoshu

**Pull request merged**